From 299d98411ff19f53242867c87075e0ae8a4dd6e6 Mon Sep 17 00:00:00 2001
From: JimmySnails <talantyyr@gmail.com>
Date: Sat, 9 May 2020 10:24:17 +0200
Subject: [PATCH] fix android cmake

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fab479d9..b82529743 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,18 @@
-cmake_minimum_required(VERSION 3.11 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 # add the path to our custom find modules
+
+# create the cmake project
+project(
+        # this will be the ${PROJECT_NAME}
+        Cytopia
+        # this will provide ${PROJECT_VERSION}
+        VERSION 0.2.1
+        # Description
+        DESCRIPTION "A city-building game"
+        # we are using C++
+        LANGUAGES CXX
+)
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
