Index: Cytopia-0.2.1/CMakeLists.txt
===================================================================
--- Cytopia-0.2.1.orig/CMakeLists.txt
+++ Cytopia-0.2.1/CMakeLists.txt
@@ -199,12 +199,6 @@ else (USE_PACKAGE_MANAGER)
 
 endif (USE_PACKAGE_MANAGER)
 
-if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  add_compile_definitions(DEBUG)
-elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
-  add_compile_definitions(DEBUG)
-endif()
-
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
   link_libraries(dbghelp.lib)
 else()
