diff --git a/utils/playerwritemap/CMakeLists.txt b/utils/playerwritemap/CMakeLists.txt
index 9f04750..e4e5389 100644
--- a/utils/playerwritemap/CMakeLists.txt
+++ b/utils/playerwritemap/CMakeLists.txt
@@ -8,10 +8,6 @@ IF (BUILD_UTILS_PLAYERWRITEMAP)
         PLAYER_ADD_EXECUTABLE (playerwritemap ${playerwritemapSrcs})
         TARGET_LINK_LIBRARIES (playerwritemap playerc playerinterface playercommon
             ${PLAYERC_EXTRA_LINK_LIBRARIES} ${GDKPIXBUF_PKG_LIBRARIES})
-        SET_SOURCE_FILES_PROPERTIES (${playerwritemapSrcs} PROPERTIES
-            COMPILE_FLAGS "${GDKPIXBUF_CFLAGS}")
-        SET_TARGET_PROPERTIES (playerwritemap PROPERTIES
-            LINK_FLAGS "${GDKPIXBUF_LINKFLAGS}")
     ELSE (WITH_GDKPIXBUF)
         MESSAGE (STATUS "playerwritemap will not be built - GDK pixbuf not found")
     ENDIF (WITH_GDKPIXBUF)
