--- a/iraspa/iraspagui/qtsegmentcontrol.cpp	2026-04-18 15:33:09.924724727 +0200
+++ b/iraspa/iraspagui/qtsegmentcontrol.cpp	2026-04-18 15:33:09.926256908 +0200
@@ -133,7 +133,7 @@
             QPixmap pm;
 
             QSize buttonSize = widget->rect().size();
-            const QString key = QString("qt_segment %0 %1 %2").arg(option->state).arg(buttonSize.width()).arg(buttonSize.height());
+            const QString key = QString("qt_segment %0 %1 %2").arg(static_cast<int>(option->state)).arg(buttonSize.width()).arg(buttonSize.height());
 
             if (!QPixmapCache::find(key, &pm)) {
                 pm = QPixmap(buttonSize);
