--- a/CMake/vtkTargets/vtkOffscreen/CMakeLists.txt
+++ b/CMake/vtkTargets/vtkOffscreen/CMakeLists.txt
@@ -83,7 +83,7 @@
     ExternalProject_Get_Property( vtkdicom-offscreen INSTALL_DIR )
     ExternalProject_Get_Property( vtkdicom-offscreen SOURCE_DIR )

-    set(VTKDICOMOFFSCREEN_INCLUDE_DIR ${INSTALL_DIR}/include/vtk-8.90)
+    set(VTKDICOMOFFSCREEN_INCLUDE_DIR ${INSTALL_DIR}/include/vtk-9.4)
     set(VTKDICOMOFFSCREEN_LIBDIR ${INSTALL_DIR}/lib)
     set(VTKDICOMOFFSCREEN_LIBRARY ${VTKDICOMOFFSCREEN_LIBDIR}/${LIBNAME})

--- a/CMake/vtkTargets/vtkOnscreen/CMakeLists.txt
+++ b/CMake/vtkTargets/vtkOnscreen/CMakeLists.txt
@@ -130,7 +130,7 @@
 if (WIN32)
   set(VTKDICOMONSCREEN_INCLUDE_DIR ${INSTALL_DIR}/include)
 else()
-  set(VTKDICOMONSCREEN_INCLUDE_DIR ${INSTALL_DIR}/include/vtk-8.90)
+  set(VTKDICOMONSCREEN_INCLUDE_DIR ${INSTALL_DIR}/include/vtk-9.4)
 endif()
 set(VTKDICOMONSCREEN_LIBDIR ${INSTALL_DIR}/lib)
 set(VTKDICOMONSCREEN_LIBRARY ${VTKDICOMONSCREEN_LIBDIR}/${LIBNAME})
