--- a/OMEdit/OMEditLIB/OMEditApplication.cpp
+++ b/OMEdit/OMEditLIB/OMEditApplication.cpp
@@ -165,7 +165,7 @@
       // install application translations
       QString translationsLocation = installationDirectoryPath + QString("/share/omedit/nls");
       // skip loading OMEdit translation file if locale language is QLocale::English. We don't have any OMEdit_en*.qm file.
-      if (settingsLocale.language() != QLocale::English) {
+      if (QLocale(locale).language() != QLocale::English) {
         if (mTranslator.load("OMEdit_" + locale, translationsLocation)) {
           installTranslator(&mTranslator);
         } else {
